synapse/handlers/pagination.py

```diff
@@ -49,19 +49,31 @@ class PurgeStatus:
     STATUS_ACTIVE = 0
     STATUS_COMPLETE = 1
     STATUS_FAILED = 2
+    STATUS_REMOVE_MEMBERS = 3
 
     STATUS_TEXT = {
         STATUS_ACTIVE: "active",
         STATUS_COMPLETE: "complete",
         STATUS_FAILED: "failed",
+        STATUS_REMOVE_MEMBERS: "remove members",
     }
 
-    # Tracks whether this request has completed. One of STATUS_{ACTIVE,COMPLETE,FAILED}.
+    # Tracks whether this request has completed.
+    # One of STATUS_{ACTIVE,COMPLETE,FAILED, STATUS_REMOVE_MEMBERS}.
     status: int = STATUS_ACTIVE
 
+    # Saves the result of an action to give it back to REST API
+    result: Dict = {}
+
     def asdict(self) -> JsonDict:
         return {"status": PurgeStatus.STATUS_TEXT[self.status]}
 
+    def asdict_with_result(self) -> JsonDict:
+        return {
+            "status": PurgeStatus.STATUS_TEXT[self.status],
+            "result": self.result,
+        }
```

Comment:
Yes, sure.
But then `/purge_histrory_status` gets always this information:
```json
{
  "kicked_users": [],
  "failed_to_kick_users": [],
  "local_aliases": [],
  "new_room_id": null
}
```
Purging does not kick someone. This informations are always empty, confusing and does not help the user.

Reply:
does it though? I thought it would get an empty dict.

Anyway, there's nothing stopping a caller using `asdict` on a delete request. It shouldn't just strip the data out.